core/src/main/scala/org/http4s/internal/parsing/Rfc2616.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2013-2020 http4s.org
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.http4s.internal.parsing
+
+import cats.parse.Parser1
+import cats.parse.Parser.{char, charIn, string1}
+import cats.parse.Rfc5234.{digit, sp}
+
+/** Common rules defined in RFC2616.  This RFC is now obsolete,
+  * but some other active ones still refer to it.
+  *
+  * @see [[https://tools.ietf.org/html/rfc2616#section-2.2] RFC2616, Basic Rules]
+  */
+private[http4s] object Rfc2616 {
+  /* CHAR           = <any US-ASCII character (octets 0 - 127)>
+   * CTL            = <any US-ASCII control character
+   *                  (octets 0 - 31) and DEL (127)>
+   * separators     = "(" | ")" | "<" | ">" | "@"
+   *                | "," | ";" | ":" | "\" | <">
+   *                | "/" | "[" | "]" | "?" | "="
+   *                | "{" | "}" | SP  | HT
+   * token          = 1*<any CHAR except CTLs or separators>
+   */
+  val token: Parser1[String] = {
+    val ascii = Set(0.toChar to 127.toChar: _*)
+    val ctl = Set(0.toChar to 31.toChar: _*) + 127.toChar
+    val separators = Set("()<>@,;:\\\"/[]?={} \t".toSeq: _*)
+
+    val tchar = charIn(ascii -- ctl -- separators)
+    tchar.rep1.string
+  }
+
+  final case class Rfc1123Date(
+      wkday: Int,
+      day: Int,
+      month: Int,
+      year: Int,
+      hour: Int,
+      min: Int,
+      sec: Int)
+
+  val rfc1123Date: Parser1[Rfc1123Date] = {
```

Comment:
This whole parser is redundant with `HttpDate`, and brings up an interesting case of having two specs parsing to the same type.  We could move this onto `HttpDate` under a different name.  This is one of the first types "parser on the companion" vs. "parser with the rest of the RFC" came into tension.

Reply:
I think it's also redundant with one of the new parsers, but we've at least traced it to the spec.